src/lib/y2storage/proposal.rb

```diff
@@ -78,91 +78,94 @@ def propose
       raise UnexpectedCallError if proposed?
       settings.freeze
       @proposed = true
-      @volumes = volumes_list(:all, populated_settings)
-      @devices = devicegraph(@volumes, populated_settings)
+
+      # FIXME: The current implementation, tries :desired and then :min for
+      # every root disk. It would probably make more sense to try first :desired
+      # for all possible root disks and then do the same with :min.
+      exception = nil
+      candidate_roots.each do |disk_name|
+        populated_settings.root_device = disk_name
+        exception = nil
+
+        begin
+          @volumes = volumes_list
+          @devices = devicegraph(@volumes)
+        rescue Error => error
+          log.info "Failed to make a proposal using root device #{disk_name}: #{error.message}"
+          exception = error
+        end
+
+        break unless exception
```

Comment:
well, OK. Even I don't find it natural. Especially to return only exception of the last candidate when all candidates failed. But not critical.